iocore/cache/P_CacheHosting.h

```diff
@@ -163,8 +163,9 @@ struct CacheHostTableConfig : public Continuation {
   {
     (void)e;
     (void)event;
-    CacheHostTable *t   = new CacheHostTable((*ppt)->cache, (*ppt)->type);
-    CacheHostTable *old = (CacheHostTable *)ink_atomic_swap(&t, *ppt);
+    CacheHostTable *t = new CacheHostTable((*ppt)->cache, (*ppt)->type);
+    auto old          = *ppt;
+    (*ppt)            = t;
```

Comment:
I think this needs to be a `std::atomic<CacheHostTable *>* ppt` or possibly `std::atomic<std::atomic<CacheHostTable *>*> ppt;`.

With the outer pointer-pointer being atomic, but not the inner, I believe in `auto old = *ppt; the dereference `*ppt` will be atomic, but not the assignment to `old`. Likewise, in `(*ppt) = t;` I believe the `*ppt` dereference will be atomic, but when it takes `t` and assigns it to the value-pointer-to-by `ppt`, I don't think that will be atomic.